src/Dialect/ONNX/ShapeInference/RoiAlign.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===-------------- RoiAlign.cpp - Shape Inference for RoiAlign Op
+//--------------===//
+//
+// This file implements shape inference for the ONNX Shape Operator.
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/Dialect/ONNX/ShapeInference/ONNXShapeHelper.hpp"
+
+ONNXRoiAlignOpShapeHelper::ONNXRoiAlignOpShapeHelper(ONNXRoiAlignOp *newOp)
+    : ONNXOpShapeHelper<ONNXRoiAlignOp>(
+          newOp, newOp->getOperation()->getNumResults()),
+      xDims(), batchIndicesDims() {}
+
+ONNXRoiAlignOpShapeHelper::ONNXRoiAlignOpShapeHelper(ONNXRoiAlignOp *newOp,
+    OpBuilder *rewriter, ArrayValueIndexCapture::GetDenseVal fGetDenseVal,
+    ArrayValueIndexCapture::LoadVal fLoadVal)
+    : ONNXOpShapeHelper<ONNXRoiAlignOp>(newOp,
+          newOp->getOperation()->getNumResults(), rewriter, fGetDenseVal,
+          fLoadVal),
+      xDims(), batchIndicesDims() {}
+
+LogicalResult ONNXRoiAlignOpShapeHelper::computeShape(
+    ONNXRoiAlignOpAdaptor operandAdaptor) {
+  // Shape inference indicated by passing a null rewriter pointer.
+  // get input info.
+  Value X = operandAdaptor.X();
+  Value batch_indices = operandAdaptor.batch_indices();
+
+  // Test ranks.
+  if (X.getType().cast<ShapedType>().getShape().size() != 4)
```

Comment:
The preferred way is to have a verifier that isolate all of the testing, and a shape inference that does  the shape inferencing. See HowToAddAnOperation.md for help on how to do it

Reply:
A simple test case is added.

Reply:
Thanks for adding the example. Are you willing to move the "testing for error" into a separate verifier? That is the preferred solution. Easiest to do it while adding the code, though it can be migrated later if you don't have the time now.